File: src/main/java/bad/robot/radiate/teamcity/RunningBuild.java
package bad.robot.radiate.teamcity;

import bad.robot.radiate.Activity;
import bad.robot.radiate.Progress;
import com.google.gson.annotations.SerializedName;

import static bad.robot.radiate.Activity.Progressing;

public class RunningBuild extends Build {

    @SerializedName("running-info")
    private final RunInformation runInformation;

    public RunningBuild(String id, String number, String href, String status, String statusText, String startDate, String finishDate, BuildType buildType, RunInformation runInformation) {
        super(id, number, href, status, statusText, startDate, finishDate, buildType);
        this.runInformation = runInformation;
    }

    public RunInformation getRunInformation() {
        return runInformation;
    }

    @Override
    public Activity getActivity() {
        return Progressing;
    }

    @Override
    public Progress getProgress() {
        return new Progress(runInformation.getPercentageComplete(), 100);
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/RunningBuild.scala
package bad.robot.radiate.teamcity

import bad.robot.radiate.{ProgressS, Progressing}
import com.google.gson.annotations.SerializedName

class RunningBuildS(id: String, number: String, href: String, status: String, statusText: String, startDate: String, finishDate: String, buildType: BuildTypeScala, @SerializedName("running-info") private val _runInformation: RunInformation) extends BuildS(id, number, href, status, statusText, startDate, finishDate, buildType) {

  def runInformation = _runInformation

  override def activity = Progressing

  override def progress = new ProgressS(runInformation.getPercentageComplete, 100)
}
